lightning/src/ln/channelmanager.rs

```diff
@@ -192,6 +192,13 @@ pub(super) enum HTLCForwardInfo {
 	},
 }
 
+// Used for failing blinded HTLCs backwards correctly.
+#[derive(Clone, Debug, Hash, PartialEq, Eq)]
+enum Blinded {
```

Comment:
`BlindType` (or `Category`)? `Origin` sounds like the sender to me.

Reply:
Maybe `BlindedErrorPolicy::{Non}IntroductionNode`? Like it refers to the plan for how we'll error if the htlc fails? Bit long but...

Reply:
Will it basically be `BlindedForward` without any data? I was about to suggest that name but it is taken lol. Would a `bool` suffice? Or at least a name similar to `BlindedForward` but for failing backwards?

Reply:
I had it as a bool previously but Matt wanted an enum because we'd have to store it in `HTLCPreviousHopData` as `blinded: Option<bool>` which is a bit weird. 

`BlindedForwardFailure`? Except we'll use it for received htlcs too. `BlindedFailureType`?

Reply:
Discussed offline, going with `BlindedFailure::From{Non}IntroductionNode`.